### PR TITLE
Faster date highlight/selection

### PR DESCRIPTION
**Pull request by edenman**
Built on top of #338

### CODE REVIEW

**Comment by edenman** [comment]
@JakeWharton can you give me a sanity check on `IndexedLinkedHashMap`?

File: library/src/main/java/com/squareup/timessquare/IndexedLinkedHashMap.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.Map;
+
+/**
+ * Take advantage of LinkedHashMap's iterable ordering but also keep track of the indexes and allow
+ * - Fast key lookup by index
+ * - Fast index lookup by key
+ */
+class IndexedLinkedHashMap<K, V> extends LinkedHashMap<K, V> {
+  private final Map<Integer, K> indexToKey = new LinkedHashMap<>();
+  private final Map<K, Integer> keyToIndex = new LinkedHashMap<>();
```
Comment (JakeWharton) [inline library/src/main/java/com/squareup/timessquare/IndexedLinkedHashMap.java]: These two maps can be replaced with a `SparseArray` which will save on storage.

Comment (edenman) [inline library/src/main/java/com/squareup/timessquare/IndexedLinkedHashMap.java]: So I just tried this, and unfortunately it breaks on my `getIndexOfKey` method...SparseArray uses object equality, so `indexLookup.indexOfValue(key)` returns -1.  I could iterate over all the keys but that seems worse.  Thoughts?

Diff: https://github.com/square/android-times-square/compare/edenman/faster-highlight...edenman/faster-highlight-with-sparsearray

Comment (edenman) [inline library/src/main/java/com/squareup/timessquare/IndexedLinkedHashMap.java]: where i found out that SparseArray uses object equality: https://code.google.com/p/android/issues/detail?id=53297

**Comment by FireZenk** [comment]
When are you going to merge this?

**Comment by edenman** [comment]
Thanks for the prod, @FireZenk, I had totally forgotten about this PR.  And thanks again for the original PR!

**Comment by edenman** [comment]
Just released this in 1.7.1, should be live in a couple hours

**Pull request merged**